pkg/sql/evalcatalog/pg_updatable.go

```diff
@@ -33,6 +37,25 @@ var (
 	allUpdatableEvents = tree.NewDInt((1 << UpdateCommand) | (1 << InsertCommand) | (1 << DeleteCommand))
 )
 
+// RedactDescriptor takes an encoded protobuf descriptor and returns the
+// encoded protobuf after redaction.
+func (b *Builtins) RedactDescriptor(ctx context.Context, encodedDescriptor []byte) ([]byte, error) {
+	var desc descpb.Descriptor
+	if err := protoutil.Unmarshal(encodedDescriptor, &desc); err != nil {
+		return nil, err
+	}
+	builder := descbuilder.NewBuilder(&desc)
```

Comment:
This line is causing a panic/node crash when I run:
```
SELECT id, crdb_internal.redact_descriptor(descriptor) FROM system.descriptor
```

Stack trace:
```
goroutine 1035 [running]:
github.com/cockroachdb/cockroach/pkg/util/log.getStacks(0x0)
	github.com/cockroachdb/cockroach/pkg/util/log/get_stacks.go:25 +0x89
github.com/cockroachdb/cockroach/pkg/util/log.(*loggerT).outputLogEntry(0xc001f8b0c0, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}}, 0x171433b0167723b8, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/clog.go:260 +0xb8
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0xa0a6708, 0xc00012a008}, 0x2, 0x4, 0x0, 0x0?, {0x8e0ab52, 0x68}, {0xc001955328, 0x3, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:106 +0x645
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
github.com/cockroachdb/cockroach/pkg/util/log.Fatalf(...)
	github.com/cockroachdb/cockroach/bazel-out/darwin-fastbuild/bin/pkg/util/log/log_channels_generated.go:848
github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb.MaybeSetDescriptorModificationTimeFromMVCCTimestamp(0xc001a48b20, {0x3ac?, 0x3?, 0x0?})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb/pkg/sql/catalog/descpb/descriptor.go:194 +0x595
github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb.FromDescriptorWithMVCCTimestamp(0x10?, {0xe6005b8?, 0x10?, 0x0?})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb/pkg/sql/catalog/descpb/descriptor.go:234 +0xb6
github.com/cockroachdb/cockroach/pkg/sql/catalog/descbuilder.NewBuilderWithMVCCTimestamp(0xc0013bec00?, {0x3ac?, 0x400?, 0x0?})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/descbuilder/desc_builder.go:32 +0x1d
github.com/cockroachdb/cockroach/pkg/sql/catalog/descbuilder.NewBuilder(...)
	github.com/cockroachdb/cockroach/pkg/sql/catalog/descbuilder/desc_builder.go:56
github.com/cockroachdb/cockroach/pkg/sql/evalcatalog.(*Builtins).RedactDescriptor(0x0?, {0xc0024df000?, 0x3ac?}, {0xc0013bec00, 0x3ac, 0x400})
	github.com/cockroachdb/cockroach/pkg/sql/evalcatalog/pg_updatable.go:48 +0x9d
```

Looking into it now, will update once I find more info.

Reply:
There's some annoying logic in there that relies on the MVCC Timestamp, perhaps that should be an argument to the builtin actually.

Reply:
We're hitting [this case here](https://github.com/cockroachdb/cockroach/blob/e1fa2122da04e1aba6ce76f69fef524cc099e967/pkg/sql/catalog/descpb/descriptor.go#L190-L195).

The logged error & descriptor:
```
F220912 19:29:53.790723 1035 sql/catalog/descpb/descriptor.go:194 ⋮ [-] 83  read table descriptor for ‹"warehouse"› (109) without ModificationTime with zero MVCC timestamp; full descriptor:
F220912 19:29:53.790723 1035 sql/catalog/descpb/descriptor.go:194 ⋮ [-] 83 +‹table:<name:"warehouse" id:109 version:8 modification_time:<> parent_id:107 unexposed_parent_schema_id:108 columns:<name:"w_id" id:1 type:<family: IntFamily width: 64 precision: 0 locale: "" visible_type: 0 oid: 20 time_precision_is_set: false > nullable:false hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > columns:<name:"w_name" id:2 type:<family: StringFamily width: 10 precision: 0 locale: "" visible_type: 0 oid: 1043 time_precision_is_set: false > nullable:false hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > columns:<name:"w_street_1" id:3 type:<family: StringFamily width: 20 precision: 0 locale: "" visible_type: 0 oid: 1043 time_precision_is_set: false > nullable:false hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > columns:<name:"w_street_2" id:4 type:<family: StringFamily width: 20 precision: 0 locale: "" visible_type: 0 oid: 1043 time_precision_is_set: false > nullable:false hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > columns:<name:"w_city" id:5 type:<family: StringFamily width: 20 precision: 0 locale: "" visible_type: 0 oid: 1043 time_precision_is_set: false > nullable:false hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > columns:<name:"w_state" id:6 type:<family: StringFamily width: 2 precision: 0 locale: "" visible_type: 0 oid: 1042 time_precision_is_set: false > nullable:false hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > columns:<name:"w_zip" id:7 type:<family: StringFamily width: 9 precision: 0 locale: "" visible_type: 0 oid: 1042 time_precision_is_set: false > nullable:false hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > columns:<name:"w_tax" id:8 type:<family: DecimalFamily width: 4 precision: 4 locale: "" visible_type: 0 oid: 1700 time_precision_is_set: false > nullable:false hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > columns:<name:"w_ytd" id:9 type:<family: DecimalFamily width: 2 precision: 12 locale: "" visible_type: 0 oid: 1700 time_precision_is_set: false > nullable:false hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > next_column_id:10 families:<name:"primary" id:0 column_names:"w_id" column_names:"w_name" column_names:"w_street_1" column_names:"w_street_2" column_names:"w_city" column_names:"w_state" column_names:"w_zip" column_names:"w_tax" column_names:"w_ytd" column_ids:1 column_ids:2 column_ids:3 column_ids:4 column_ids:5 column_ids:6 column_ids:7 column_ids:8 column_ids:9 default_column_id:0 > next_family_id:1 primary_index:<name:"warehouse_pkey" id:1 unique:true version:4 key_column_names:"w_id" key_column_directions:ASC store_column_names:"w_name" store_column_names:"w_street_1" store_column_names:"w_street_2" store_column_names:"w_city" store_column_names:"w_state" store_column_names:"w_zip" store_column_names:"w_tax" store_column_names:"w_ytd" key_column_ids:1 store_column_ids:2 store_column_ids:3 store_column_ids:4 store_column_ids:5 store_column_ids:6 store_column_ids:7 store_column_ids:8 store_column_ids:9 foreign_key:<table:0 index:0 name:"" validity:Validated shared_prefix_len:0 on_delete:NO_ACTION on_update:NO_ACTION match:SIMPLE > interleave:<> partitioning:<num_columns:0 num_implicit_columns:0 > type:FORWARD created_explicitly:false encoding_type:1 sharded:<is_sharded:false name:"" shard_buckets:0 > disabled:false geo_config:<> predicate:"" use_delete_preserving_encoding:false created_at_nanos:1663010571976328000 constraint_id:1 not_visible:false > next_index_id:2 privileges:<users:<user_proto:"admin" privileges:2 with_grant_option:2 > users:<user_proto:"root" privileges:2 with_grant_option:2 > owner_proto:"root" version:2 > next_mutation_id:1 format_version:3 state:PUBLIC offline_reason:"" view_query:"" is_materialized_view:false refresh_view_required:false drop_time:0 replacement_of:<id:0 time:<> > audit_mode:DISABLED drop_job_id:0 create_query:"" create_as_of_time:<wall_time:1663010571976333000 > inbound_fks:<origin_table_id:110 origin_column_ids:2 referenced_column_ids:1 referenced_table_id:109 name:"district_d_w_id_fkey" validity:Unvalidated on_delete:NO_ACTION on_update:NO_ACTION match:SIMPLE constraint_id:2 > inbound_fks:<origin_table_id:116 origin_column_ids:2 referenced_column_ids:1 referenced_table_id:109 name:"stock_s_w_id_fkey" validity:Unvalidated on_delete:NO_ACTION on_update:NO_ACTION match:SIMPLE constraint_id:2 > temporary:false partition_all_by:false exclude_data_from_backup:false next_constraint_id:2 import_start_wall_time:0 >›
```

Reply:
I'm hoping https://github.com/cockroachdb/cockroach/pull/88786 will fix this. At the very least, there won't be a panic, so that's that.